src/main/java/io/fabric8/elasticsearch/plugin/DynamicACLFilter.java

```diff
@@ -0,0 +1,161 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.elasticsearch.plugin;
+
+import java.io.IOException;
+import java.util.Set;
+
+import org.apache.commons.lang.ObjectUtils;
+import org.apache.commons.lang.StringUtils;
+import org.elasticsearch.action.get.GetResponse;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.common.logging.ESLogger;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.rest.RestChannel;
+import org.elasticsearch.rest.RestFilter;
+import org.elasticsearch.rest.RestFilterChain;
+import org.elasticsearch.rest.RestRequest;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.fabric8.elasticsearch.plugin.acl.SearchGuardACL;
+import io.fabric8.elasticsearch.plugin.osclient.Fabric8OpenShiftClientAdapter;
+import io.fabric8.elasticsearch.plugin.osclient.OpenShiftClient;
+import io.fabric8.kubernetes.client.internal.Utils;
+
+public class DynamicACLFilter extends RestFilter {
+
+	private static final String AUTHORIZATION_HEADER = "Authorization";
+	private static final String DEFAULT_SECURITY_CONFIG_INDEX = "searchguard";
+	private static final String DEFAULT_AUTH_PROXY_HEADER = "X-Authenticated-User";
+	private static final String SEARCHGUARD_TYPE = "ac";
+	private static final String MASTER_API_URL = "K8S_HOST_URL";
+	private static final String CERT_CA_FILE = "KUBERNETES_CERTS_CA_FILE"; 
+	private static final String TRUST_CERTS = "KUBERNETES_TRUST_CERTIFICATES"; 
+	private static final String SEARCHGUARD_ID = "ac";
+	private static final String SEARCHGUARD_AUTHENTICATION_PROXY_HEADER = "searchguard.authentication.proxy.header";
+	private static final String SEARCHGUARD_CONFIG_INDEX_NAME = "searchguard.config_index_name";
+
+
+	private final ObjectMapper mapper = new ObjectMapper();
+	private final ESLogger logger;
+	private final UserProjectCache cache = new UserProjectCacheMapAdapter();
+	private final String proxyUserHeader;
+	private final Client esClient;
+	private final String searchGuardIndex;
+	private final String masterApiUrl;
+	private final String caCertFile;
+	private final Boolean trustCerts;
+
+	public DynamicACLFilter(final Settings settings, final Client client, final ESLogger logger){
+		this.logger = logger;
+		this.esClient = client;
+		this.proxyUserHeader = settings.get(SEARCHGUARD_AUTHENTICATION_PROXY_HEADER, DEFAULT_AUTH_PROXY_HEADER);
+		this.searchGuardIndex = settings.get(SEARCHGUARD_CONFIG_INDEX_NAME, DEFAULT_SECURITY_CONFIG_INDEX);
+		this.masterApiUrl = Utils.getSystemPropertyOrEnvVar(MASTER_API_URL);
```

Comment:
Client uses `KUBERNETES_MASTER` env var & defaults to `kubernetes.default.svc` so it works in-cluster without any config. Remove.